src/modules/Bootloader.rb

```diff
@@ -166,8 +166,14 @@ def Read
 
         ::Bootloader::BootloaderFactory.current = ::Bootloader::BootloaderFactory.proposed
         ::Bootloader::BootloaderFactory.current.propose
+      rescue ::Bootloader::InsufficientPrivileges => e
```

Comment:
## A self pre-review / Request for comments

I added this new exception just to follow the pattern used here (I mean, rescuing its own exception). But, to be honest, I'm not sure if

* I like it
* It is a good idea

The other way I see to do it without a custom exception is to simply rescue the `Errno::ACCES` here directly.

```ruby

...
rescue Errno::ACCES
  error_msg = _(
    "The module is running without enough privileges to perform all possible actions.\n\n" \
    "Cannot continue. Please, try again as root."
  )

  Yast2::Popup.show(error_msg, headline: :error)

  return false
end

...
```

Shorter than the previous option but, again, I'm not sure if

* I like it
* It is a good idea

**What do you think? Do you have any other suggestions?**

Thanks in advance for your input!

Reply:
I agree there is no need to invent a new type of exception that is only used to directly replace `Errno::ACCESS`.

The current custom exceptions defined in yast2-bootloader exist because they actually describe a pretty specific circumstance that is not covered by the standard ruby. But that's not the case with that new (and unnecessary) `InsufficientPrivileges` class.